test/browser-mananger.test.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const assert = require('assert').strict;
+const BrowserManager = require('../lib/browser-mananger.js');
+
+describe('BrowserManager', () => {
+    it('should handle multiple instances creation', async() => {
+        const browserManager = new BrowserManager();
+        const [instance1, instance2] = await Promise.all([
+            browserManager.getBrowser(),
+            browserManager.getBrowser()
+        ]);
+
+        assert.ok(instance1);
+        assert.ok(instance2);
+        assert.deepEqual(instance1, instance2);
+
+        await browserManager.closeBrowser();
+    });
+
+    it('should not throw an exception when the instance does not exist but calls closeBrowser', () => {
+        const browserManager = new BrowserManager();
```

Comment:
Some tests are missing `assert` [stage](https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/) which makes tests less great and may cause problems on CI machine.
Can you please use `assert.doesNotThrow` ? And it will be great to specify expecting error if possible

Reply:
These tests just validate the concurrency management. You can add assert. the `closeBrowser()` is an async method so I think the `assert.doesNotReject()` is more proper

Reply:
I don't expect any exception